devdoc/EndUsers.md

```diff
@@ -2,14 +2,115 @@
+
+[Helm Examples](./helmReadmes.md) General documentation for the Helm examples.
+
 ## Dali
 
-## Sasha
+Dali is the central metadata repository and coordination service in the HPCC Systems Platform. It plays a critical role in managing and storing system metadata, including file information, cluster configurations, and workunit details. Dali ensures consistency and synchronization across the platform, enabling efficient resource management and job execution.
 
 ## DFU
 
+The Distributed File Utility (DFU) is a core component of the HPCC Systems Platform that facilitates the management of files across the distributed environment. It provides tools for transferring, replicating, and managing data efficiently. DFU ensures data integrity and supports operations such as file copy, delete, spray (import), and despray (export).
+
+### Key Features of DFU
+
+- **File Transfer (Copy/Remote Copy)**: Move files between nodes or clusters seamlessly.
+- **Data Spray**: Import large datasets into the HPCC Systems Platform for processing.
+- **Data Despray**: Export processed data from the platform to external systems.
+- **Replication**: Ensure data redundancy and availability across the cluster.
+- **File Management**: Perform operations like renaming, deleting, or modifying file attributes.
+- **Superfile Management**: Perform superfile operations like adding subfiles, deleting subfiles, or modifying file attributes.
+
 ## DFUPlus
 
+DFUPlus is a command-line utility that provides advanced file management capabilities within the HPCC Systems Platform. It is designed to interact with the Distributed File Utility (DFU) service, enabling users to perform file operations programmatically or through scripts.
+
+[Client Tools Documentation](https://hpccsystems.com/training/documentation/ecl-ide-and-client-tools/)
+
+## ECL
+
+### ECL Language Reference
+
+ECL is the Enterprise Control Language designed specifically for huge data projects using the HPCC Systems platform. Its extreme scalability comes from a design that allows you to leverage every query you create for re-use in subsequent queries as needed.
+
+[ECL Language Reference](https://hpccsystems.com/wp-content/uploads/_documents/ECLR_EN_US/index.html) Online Access to the ECL Language Reference.
+
+### ECL Standard Library
+
+The ECL Standard Library is a collection of pre-built functions, modules, and utilities that provide common functionality for ECL programming in the HPCC Systems Platform. It's designed to help developers write ECL code more efficiently by providing reusable components for common data processing tasks.
+
+[ECL Standard Library Reference](https://hpccsystems.com/wp-content/uploads/_documents/SLR_EN_US/index.html) Online Access to the ECL Standard Library Reference.
+
+### ECL IDE
+
+The ECL IDE is the simple and easy way to create Queries into your data, and ECL files with which to build your queries.
+
```

Comment:
remove blank line for consistency

Reply:
Headings should be surrounded by blank lines.